recipes/sdl2_image/2.0.5/conanfile.py

```diff
@@ -0,0 +1,126 @@
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "bmp": True,
+        "gif": True,
+        "lbm": True,
+        "pcx": True,
+        "pnm": True,
+        "svg": True,
+        "tga": True,
+        "xcf": True,
+        "xpm": True,
+        "xv": True,
+        "jpg": True,
+        "tif": True,
+        "png": True,
+        "webp": True,
+        "imageio": False
+    }
+
+    _cmake = None
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Macos":
+            del self.options.imageio
+
+    def requirements(self):
+        self.requires("sdl2/2.0.14@bincrafters/stable")
+        if self.options.tif:
+            self.requires("libtiff/4.0.9")
+        if self.options.jpg:
+            self.requires("libjpeg/9d")
+        if self.options.png:
+            self.requires("libpng/1.6.37")
+        if self.options.webp:
+            self.requires("libwebp/1.0.3")
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "SDL2_image-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BMP"] = self.options.bmp
+        self._cmake.definitions["GIF"] = self.options.gif
+        self._cmake.definitions["IMAGEIO"] = self.options.get_safe("imageio")
+        self._cmake.definitions["JPG"] = self.options.jpg
+        self._cmake.definitions["LBM"] = self.options.lbm
+        self._cmake.definitions["PCX"] = self.options.pcx
+        self._cmake.definitions["PNG"] = self.options.png
+        self._cmake.definitions["PNM"] = self.options.pnm
+        self._cmake.definitions["SVG"] = self.options.svg
+        self._cmake.definitions["TGA"] = self.options.tga
+        self._cmake.definitions["TIF"] = self.options.tif
+        self._cmake.definitions["WEBP"] = self.options.webp
+        self._cmake.definitions["XCF"] = self.options.xcf
+        self._cmake.definitions["XPM"] = self.options.xpm
+        self._cmake.definitions["XV"] = self.options.xv
+        self._cmake.definitions["TIF_DYNAMIC"] = self.options["libtiff"].shared if self.options.tif else False
+        self._cmake.definitions["JPG_DYNAMIC"] = self.options["libjpeg"].shared if self.options.jpg else False
+        self._cmake.definitions["PNG_DYNAMIC"] = self.options["libpng"].shared if self.options.png else False
+        self._cmake.definitions["WEBP_DYNAMIC"] = self.options["libwebp"].shared if self.options.webp else False
+        self._cmake.definitions["SDL2_DYNAMIC"] = self.options["sdl2"].shared
```

Comment:
It's not because libtiff is shared that consumer wants dynamic load. In upstream `configure.in`, there is a specific option (true by default) for each dependency for which sdl2_image supports dynamic load.
So, you can link to static, to shared, or dynamically load shared.

Moreover, there is no dynamic load of sdl2 in sdl2_image, it's always linked.

Reply:
Thanks for the catch. We are currently using a custom CMake file for sdl2_image, only the next version will have official CMake support.

I will rename the `SDL2_DYNAMIC` option name as it only seems to make a switch between static and shared linking. Not entirely sure how to proceed with the rest. 

Is dynamic loading just working, despite our custom CMake file? 
If yes, we don't necessarily need special handling I guess.
If not, it doesn't work right now

Reply:
From what I see in this custom CMake file, nothing is done with these definitions, so what is written in conanfile.py is misleading for future maintenance.